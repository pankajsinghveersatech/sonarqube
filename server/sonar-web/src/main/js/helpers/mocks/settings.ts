/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { Setting, SettingCategoryDefinition, SettingWithCategory } from '../../types/settings';

export function mockDefinition(
  overrides: Partial<SettingCategoryDefinition> = {}
): SettingCategoryDefinition {
  return {
    key: 'foo',
    category: 'foo category',
    fields: [],
    options: [],
    subCategory: 'foo subCat',
    ...overrides
  };
}

export function mockSetting(overrides: Partial<Setting> = {}): Setting {
  return {
    key: 'foo',
    value: '42',
    inherited: true,
    definition: {
      key: 'foo',
      name: 'Foo setting',
      description: 'When Foo then Bar',
      type: 'INTEGER',
      options: []
    },
    ...overrides
  };
}

export function mockSettingWithCategory(
  overrides: Partial<SettingWithCategory> = {}
): SettingWithCategory {
  return {
    key: 'foo',
    value: '42',
    inherited: true,
    definition: {
      key: 'foo',
      name: 'Foo setting',
      description: 'When Foo then Bar',
      type: 'INTEGER',
      options: [],
      category: 'general',
      fields: [],
      subCategory: 'email'
    },
    ...overrides
  };
}
