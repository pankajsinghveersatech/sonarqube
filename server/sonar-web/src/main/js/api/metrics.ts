/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import throwGlobalError from '../app/utils/throwGlobalError';
import { getJSON } from '../helpers/request';

export interface MetricsResponse {
  metrics: T.Metric[];
  p: number;
  ps: number;
  total: number;
}

export function getMetrics(data?: {
  isCustom?: boolean;
  p?: number;
  ps?: number;
}): Promise<MetricsResponse> {
  return getJSON('/api/metrics/search', data).catch(throwGlobalError);
}

export function getAllMetrics(data?: {
  isCustom?: boolean;
  p?: number;
  ps?: number;
}): Promise<T.Metric[]> {
  return inner(data);

  function inner(
    data: { p?: number; ps?: number } = { ps: 500 },
    prev?: MetricsResponse
  ): Promise<T.Metric[]> {
    return getMetrics(data).then(r => {
      const result = prev ? prev.metrics.concat(r.metrics) : r.metrics;
      if (r.p * r.ps >= r.total) {
        return result;
      }
      return inner({ ...data, p: r.p + 1 }, { ...r, metrics: result });
    });
  }
}
