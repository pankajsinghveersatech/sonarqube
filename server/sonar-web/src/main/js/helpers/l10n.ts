/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { fetchL10nBundle } from '../api/l10n';
import { L10nBundle, L10nBundleRequestParams } from '../types/l10n';
import { toNotSoISOString } from './dates';
import { get as loadFromLocalStorage, save as saveInLocalStorage } from './storage';

export type Messages = T.Dict<string>;

export const DEFAULT_LOCALE = 'en';
export const DEFAULT_MESSAGES = {
  // eslint-disable-next-line camelcase
  default_error_message: 'The request cannot be processed. Try again later.'
};

let allMessages: Messages = {};
let locale: string | undefined;

export function translate(...keys: string[]): string {
  const messageKey = keys.join('.');
  const l10nMessages = getMessages();
  if (process.env.NODE_ENV === 'development' && !l10nMessages[messageKey]) {
    // eslint-disable-next-line no-console
    console.error(`No message for: ${messageKey}`);
  }
  return l10nMessages[messageKey] || messageKey;
}

export function translateWithParameters(
  messageKey: string,
  ...parameters: Array<string | number>
): string {
  const message = getMessages()[messageKey];
  if (message) {
    return parameters
      .map(parameter => String(parameter))
      .reduce((acc, parameter, index) => acc.replace(`{${index}}`, () => parameter), message);
  }
  if (process.env.NODE_ENV === 'development') {
    // eslint-disable-next-line no-console
    console.error(`No message for: ${messageKey}`);
  }
  return `${messageKey}.${parameters.join('.')}`;
}

export function hasMessage(...keys: string[]): boolean {
  const messageKey = keys.join('.');
  return getMessages()[messageKey] != null;
}

export function getMessages() {
  if (typeof allMessages === 'undefined') {
    logWarning('L10n messages are not initialized. Use default messages.');
    return DEFAULT_MESSAGES;
  }
  return allMessages;
}

export function resetMessages(newMessages: Messages) {
  allMessages = newMessages;
}

export function getLocalizedMetricName(
  metric: { key: string; name?: string },
  short = false
): string {
  const bundleKey = `metric.${metric.key}.${short ? 'short_name' : 'name'}`;
  if (hasMessage(bundleKey)) {
    return translate(bundleKey);
  } else if (short) {
    return getLocalizedMetricName(metric);
  }
  return metric.name || metric.key;
}

export function getLocalizedCategoryMetricName(metric: { key: string; name?: string }) {
  const bundleKey = `metric.${metric.key}.extra_short_name`;
  return hasMessage(bundleKey) ? translate(bundleKey) : getLocalizedMetricName(metric, true);
}

export function getLocalizedMetricDomain(domainName: string) {
  const bundleKey = `metric_domain.${domainName}`;
  return hasMessage(bundleKey) ? translate(bundleKey) : domainName;
}

export function getCurrentLocale() {
  return locale;
}

export function resetCurrentLocale(newLocale: string) {
  locale = newLocale;
}

export function getShortMonthName(index: number) {
  const months = [
    'Jan',
    'Feb',
    'Mar',
    'Apr',
    'May',
    'Jun',
    'Jul',
    'Aug',
    'Sep',
    'Oct',
    'Nov',
    'Dec'
  ];
  return translate(months[index]);
}

export function getWeekDayName(index: number) {
  const weekdays = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
  return weekdays[index] ? translate(weekdays[index]) : '';
}

export function getShortWeekDayName(index: number) {
  const weekdays = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
  return weekdays[index] ? translate(weekdays[index]) : '';
}

const L10N_BUNDLE_LS_KEY = 'l10n.bundle';

export async function loadL10nBundle() {
  const bundle = await getLatestL10nBundle().catch(() => ({
    locale: DEFAULT_LOCALE,
    messages: {}
  }));

  resetCurrentLocale(bundle.locale);
  resetMessages(bundle.messages);

  // No need to load english (default) bundle, it's coming with react-intl
  if (bundle.locale !== DEFAULT_LOCALE) {
    const [intlBundle, intl] = await Promise.all([
      import(`react-intl/locale-data/${bundle.locale}`),
      import('react-intl')
    ]);

    intl.addLocaleData(intlBundle.default);
  }

  return bundle;
}

export async function getLatestL10nBundle() {
  const browserLocale = getPreferredLanguage();
  const cachedBundle = loadL10nBundleFromLocalStorage();

  const params: L10nBundleRequestParams = {};

  if (browserLocale) {
    params.locale = browserLocale;

    if (
      cachedBundle.locale &&
      browserLocale.startsWith(cachedBundle.locale) &&
      cachedBundle.timestamp &&
      cachedBundle.messages
    ) {
      params.ts = cachedBundle.timestamp;
    }
  }

  const { effectiveLocale, messages } = await fetchL10nBundle(params).catch(response => {
    if (response && response.status === 304) {
      return {
        effectiveLocale: cachedBundle.locale || browserLocale || DEFAULT_LOCALE,
        messages: cachedBundle.messages ?? {}
      };
    }
    throw new Error(`Unexpected status code: ${response.status}`);
  });

  const bundle = {
    timestamp: toNotSoISOString(new Date()),
    locale: effectiveLocale,
    messages
  };

  saveL10nBundleToLocalStorage(bundle);

  return bundle;
}

export function getCurrentL10nBundle() {
  return loadL10nBundleFromLocalStorage();
}

function getPreferredLanguage() {
  return window.navigator.languages ? window.navigator.languages[0] : window.navigator.language;
}

function loadL10nBundleFromLocalStorage() {
  let bundle: L10nBundle;

  try {
    bundle = JSON.parse(loadFromLocalStorage(L10N_BUNDLE_LS_KEY) ?? '{}');
  } catch {
    bundle = {};
  }

  return bundle;
}

function saveL10nBundleToLocalStorage(bundle: L10nBundle) {
  saveInLocalStorage(L10N_BUNDLE_LS_KEY, JSON.stringify(bundle));
}

function logWarning(message: string) {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line no-console
    console.warn(message);
  }
}
