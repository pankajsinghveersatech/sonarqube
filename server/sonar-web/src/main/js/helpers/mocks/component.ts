/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { ComponentQualifier, TreeComponent } from '../../types/component';
import { MetricKey } from '../../types/metrics';
import { mockMeasureEnhanced } from '../testMocks';

export function mockComponent(overrides: Partial<T.Component> = {}): T.Component {
  return {
    breadcrumbs: [],
    key: 'my-project',
    name: 'MyProject',
    qualifier: ComponentQualifier.Project,
    qualityGate: { isDefault: true, key: '30', name: 'Sonar way' },
    qualityProfiles: [
      {
        deleted: false,
        key: 'my-qp',
        language: 'ts',
        name: 'Sonar way'
      }
    ],
    tags: [],
    ...overrides
  };
}

export function mockTreeComponent(overrides: Partial<TreeComponent>): TreeComponent {
  return {
    key: 'my-key',
    qualifier: ComponentQualifier.Project,
    name: 'component',
    visibility: 'public',
    ...overrides
  };
}

export function mockComponentMeasure(
  file = false,
  overrides: Partial<T.ComponentMeasure> = {}
): T.ComponentMeasure {
  if (file) {
    return {
      key: 'foo:src/index.tsx',
      name: 'index.tsx',
      qualifier: ComponentQualifier.File,
      path: 'src/index.tsx',
      measures: [{ metric: MetricKey.bugs, value: '1', bestValue: false }],
      ...overrides
    };
  }
  return {
    key: 'foo',
    name: 'Foo',
    qualifier: ComponentQualifier.Project,
    measures: [{ metric: MetricKey.bugs, value: '12', bestValue: false }],
    ...overrides
  };
}

export function mockComponentMeasureEnhanced(
  overrides: Partial<T.ComponentMeasureEnhanced> = {}
): T.ComponentMeasureEnhanced {
  return {
    ...mockComponentMeasure(false, overrides as T.ComponentMeasure),
    leak: undefined,
    measures: [mockMeasureEnhanced()],
    value: undefined,
    ...overrides
  };
}
