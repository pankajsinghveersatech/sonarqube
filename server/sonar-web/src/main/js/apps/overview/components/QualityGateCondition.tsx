/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as classNames from 'classnames';
import * as React from 'react';
import { Link } from 'react-router';
import IssueTypeIcon from '../../../components/icons/IssueTypeIcon';
import Measure from '../../../components/measure/Measure';
import DrilldownLink from '../../../components/shared/DrilldownLink';
import { getBranchLikeQuery } from '../../../helpers/branch-like';
import { translate } from '../../../helpers/l10n';
import { formatMeasure, isDiffMetric } from '../../../helpers/measures';
import {
  getComponentIssuesUrl,
  getComponentSecurityHotspotsUrl,
  Location
} from '../../../helpers/urls';
import { BranchLike } from '../../../types/branch-like';
import { IssueType } from '../../../types/issues';
import { MetricKey } from '../../../types/metrics';
import { QualityGateStatusConditionEnhanced } from '../../../types/quality-gates';

interface Props {
  branchLike?: BranchLike;
  component: Pick<T.Component, 'key'>;
  condition: QualityGateStatusConditionEnhanced;
}

export default class QualityGateCondition extends React.PureComponent<Props> {
  getIssuesUrl = (sinceLeakPeriod: boolean, customQuery: T.Dict<string>) => {
    const query: T.Dict<string | undefined> = {
      resolved: 'false',
      ...getBranchLikeQuery(this.props.branchLike),
      ...customQuery
    };
    if (sinceLeakPeriod) {
      Object.assign(query, { sinceLeakPeriod: 'true' });
    }
    return getComponentIssuesUrl(this.props.component.key, query);
  };

  getUrlForSecurityHotspot(sinceLeakPeriod: boolean) {
    const query: T.Dict<string | undefined> = {
      ...getBranchLikeQuery(this.props.branchLike)
    };
    if (sinceLeakPeriod) {
      Object.assign(query, { sinceLeakPeriod: 'true' });
    }
    return getComponentSecurityHotspotsUrl(this.props.component.key, query);
  }

  getUrlForCodeSmells(sinceLeakPeriod: boolean) {
    return this.getIssuesUrl(sinceLeakPeriod, { types: 'CODE_SMELL' });
  }

  getUrlForBugsOrVulnerabilities(type: string, sinceLeakPeriod: boolean) {
    const RATING_TO_SEVERITIES_MAPPING = [
      'BLOCKER,CRITICAL,MAJOR,MINOR',
      'BLOCKER,CRITICAL,MAJOR',
      'BLOCKER,CRITICAL',
      'BLOCKER'
    ];

    const { condition } = this.props;
    const threshold = condition.level === 'ERROR' ? condition.error : condition.warning;

    return this.getIssuesUrl(sinceLeakPeriod, {
      types: type,
      severities: RATING_TO_SEVERITIES_MAPPING[Number(threshold) - 1]
    });
  }

  wrapWithLink(children: React.ReactNode) {
    const { branchLike, component, condition } = this.props;

    const className = classNames(
      'overview-quality-gate-condition',
      `overview-quality-gate-condition-${condition.level.toLowerCase()}`
    );

    const metricKey = condition.measure.metric.key;

    const METRICS_TO_URL_MAPPING: T.Dict<() => Location> = {
      [MetricKey.reliability_rating]: () =>
        this.getUrlForBugsOrVulnerabilities(IssueType.Bug, false),
      [MetricKey.new_reliability_rating]: () =>
        this.getUrlForBugsOrVulnerabilities(IssueType.Bug, true),
      [MetricKey.security_rating]: () =>
        this.getUrlForBugsOrVulnerabilities(IssueType.Vulnerability, false),
      [MetricKey.new_security_rating]: () =>
        this.getUrlForBugsOrVulnerabilities(IssueType.Vulnerability, true),
      [MetricKey.sqale_rating]: () => this.getUrlForCodeSmells(false),
      [MetricKey.new_maintainability_rating]: () => this.getUrlForCodeSmells(true),
      [MetricKey.security_hotspots_reviewed]: () => this.getUrlForSecurityHotspot(false),
      [MetricKey.new_security_hotspots_reviewed]: () => this.getUrlForSecurityHotspot(true)
    };

    return METRICS_TO_URL_MAPPING[metricKey] ? (
      <Link className={className} to={METRICS_TO_URL_MAPPING[metricKey]()}>
        {children}
      </Link>
    ) : (
      <DrilldownLink
        branchLike={branchLike}
        className={className}
        component={component.key}
        metric={condition.measure.metric.key}
        sinceLeakPeriod={condition.period != null}>
        {children}
      </DrilldownLink>
    );
  }

  render() {
    const { condition } = this.props;
    const { measure } = condition;
    const { metric } = measure;

    const isDiff = isDiffMetric(metric.key);

    const threshold = (condition.level === 'ERROR' ? condition.error : condition.warning) as string;
    const actual = (condition.period ? measure.period?.value : measure.value) as string;

    let operator = translate('quality_gates.operator', condition.op);

    if (metric.type === 'RATING') {
      operator = translate('quality_gates.operator', condition.op, 'rating');
    }

    return this.wrapWithLink(
      <div className="overview-quality-gate-condition-container display-flex-center">
        <div className="overview-quality-gate-condition-value text-center spacer-right">
          <Measure
            decimals={2}
            metricKey={measure.metric.key}
            metricType={measure.metric.type}
            value={actual}
          />
        </div>

        <div>
          <span className="overview-quality-gate-condition-metric little-spacer-right">
            <IssueTypeIcon className="little-spacer-right" query={metric.key} />
            {metric.name}
          </span>
          {!isDiff && condition.period != null && (
            <span className="overview-quality-gate-condition-period text-ellipsis little-spacer-right">
              {translate('quality_gates.conditions.new_code')}
            </span>
          )}
          <span className="little-spacer-top small text-muted">
            {operator} {formatMeasure(threshold, metric.type)}
          </span>
        </div>
      </div>
    );
  }
}
