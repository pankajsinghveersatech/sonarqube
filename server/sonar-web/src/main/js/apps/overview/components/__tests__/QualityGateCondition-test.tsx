/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import { mockBranch } from '../../../../helpers/mocks/branch-like';
import { mockQualityGateStatusConditionEnhanced } from '../../../../helpers/mocks/quality-gates';
import { mockMetric } from '../../../../helpers/testMocks';
import { MetricKey } from '../../../../types/metrics';
import { QualityGateStatusConditionEnhanced } from '../../../../types/quality-gates';
import QualityGateCondition from '../QualityGateCondition';

it.each([
  [quickMock(MetricKey.open_issues, 'INT')],
  [quickMock(MetricKey.reliability_rating)],
  [quickMock(MetricKey.security_rating)],
  [quickMock(MetricKey.sqale_rating)],
  [quickMock(MetricKey.new_reliability_rating, 'RATING', true)],
  [quickMock(MetricKey.new_security_rating, 'RATING', true)],
  [quickMock(MetricKey.new_maintainability_rating, 'RATING', true)],
  [quickMock(MetricKey.security_hotspots_reviewed)],
  [quickMock(MetricKey.new_security_hotspots_reviewed, 'RATING', true)]
])('should render correclty', condition => {
  expect(shallowRender({ condition })).toMatchSnapshot();
});

it('should work with branch', () => {
  const condition = quickMock(MetricKey.new_maintainability_rating);
  expect(shallowRender({ branchLike: mockBranch(), condition })).toMatchSnapshot();
});

function shallowRender(props: Partial<QualityGateCondition['props']>) {
  return shallow(
    <QualityGateCondition
      component={{ key: 'abcd-key' }}
      condition={mockQualityGateStatusConditionEnhanced()}
      {...props}
    />
  );
}

function quickMock(
  metric: MetricKey,
  type = 'RATING',
  addPeriod = false
): QualityGateStatusConditionEnhanced {
  return mockQualityGateStatusConditionEnhanced({
    error: '1',
    measure: {
      metric: mockMetric({
        key: metric,
        name: metric,
        type
      }),
      value: '3',
      ...(addPeriod ? { period: { value: '3', index: 1 } } : {})
    },
    metric,
    ...(addPeriod ? { period: 1 } : {})
  });
}
