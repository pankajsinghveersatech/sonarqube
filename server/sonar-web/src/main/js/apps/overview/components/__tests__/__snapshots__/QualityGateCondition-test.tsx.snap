// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correclty 1`] = `
<DrilldownLink
  className="overview-quality-gate-condition overview-quality-gate-condition-error"
  component="abcd-key"
  metric="open_issues"
  sinceLeakPeriod={false}
>
  <div
    className="overview-quality-gate-condition-container display-flex-center"
  >
    <div
      className="overview-quality-gate-condition-value text-center spacer-right"
    >
      <Measure
        decimals={2}
        metricKey="open_issues"
        metricType="INT"
        value="3"
      />
    </div>
    <div>
      <span
        className="overview-quality-gate-condition-metric little-spacer-right"
      >
        <IssueTypeIcon
          className="little-spacer-right"
          query="open_issues"
        />
        open_issues
      </span>
      <span
        className="little-spacer-top small text-muted"
      >
        quality_gates.operator.GT
         
        1
      </span>
    </div>
  </div>
</DrilldownLink>
`;

exports[`should render correclty 2`] = `
<Link
  className="overview-quality-gate-condition overview-quality-gate-condition-error"
  onlyActiveOnIndex={false}
  style={Object {}}
  to={
    Object {
      "pathname": "/project/issues",
      "query": Object {
        "id": "abcd-key",
        "resolved": "false",
        "severities": "BLOCKER,CRITICAL,MAJOR,MINOR",
        "types": "BUG",
      },
    }
  }
>
  <div
    className="overview-quality-gate-condition-container display-flex-center"
  >
    <div
      className="overview-quality-gate-condition-value text-center spacer-right"
    >
      <Measure
        decimals={2}
        metricKey="reliability_rating"
        metricType="RATING"
        value="3"
      />
    </div>
    <div>
      <span
        className="overview-quality-gate-condition-metric little-spacer-right"
      >
        <IssueTypeIcon
          className="little-spacer-right"
          query="reliability_rating"
        />
        reliability_rating
      </span>
      <span
        className="little-spacer-top small text-muted"
      >
        quality_gates.operator.GT.rating
         
        A
      </span>
    </div>
  </div>
</Link>
`;

exports[`should render correclty 3`] = `
<Link
  className="overview-quality-gate-condition overview-quality-gate-condition-error"
  onlyActiveOnIndex={false}
  style={Object {}}
  to={
    Object {
      "pathname": "/project/issues",
      "query": Object {
        "id": "abcd-key",
        "resolved": "false",
        "severities": "BLOCKER,CRITICAL,MAJOR,MINOR",
        "types": "VULNERABILITY",
      },
    }
  }
>
  <div
    className="overview-quality-gate-condition-container display-flex-center"
  >
    <div
      className="overview-quality-gate-condition-value text-center spacer-right"
    >
      <Measure
        decimals={2}
        metricKey="security_rating"
        metricType="RATING"
        value="3"
      />
    </div>
    <div>
      <span
        className="overview-quality-gate-condition-metric little-spacer-right"
      >
        <IssueTypeIcon
          className="little-spacer-right"
          query="security_rating"
        />
        security_rating
      </span>
      <span
        className="little-spacer-top small text-muted"
      >
        quality_gates.operator.GT.rating
         
        A
      </span>
    </div>
  </div>
</Link>
`;

exports[`should render correclty 4`] = `
<Link
  className="overview-quality-gate-condition overview-quality-gate-condition-error"
  onlyActiveOnIndex={false}
  style={Object {}}
  to={
    Object {
      "pathname": "/project/issues",
      "query": Object {
        "id": "abcd-key",
        "resolved": "false",
        "types": "CODE_SMELL",
      },
    }
  }
>
  <div
    className="overview-quality-gate-condition-container display-flex-center"
  >
    <div
      className="overview-quality-gate-condition-value text-center spacer-right"
    >
      <Measure
        decimals={2}
        metricKey="sqale_rating"
        metricType="RATING"
        value="3"
      />
    </div>
    <div>
      <span
        className="overview-quality-gate-condition-metric little-spacer-right"
      >
        <IssueTypeIcon
          className="little-spacer-right"
          query="sqale_rating"
        />
        sqale_rating
      </span>
      <span
        className="little-spacer-top small text-muted"
      >
        quality_gates.operator.GT.rating
         
        A
      </span>
    </div>
  </div>
</Link>
`;

exports[`should render correclty 5`] = `
<Link
  className="overview-quality-gate-condition overview-quality-gate-condition-error"
  onlyActiveOnIndex={false}
  style={Object {}}
  to={
    Object {
      "pathname": "/project/issues",
      "query": Object {
        "id": "abcd-key",
        "resolved": "false",
        "severities": "BLOCKER,CRITICAL,MAJOR,MINOR",
        "sinceLeakPeriod": "true",
        "types": "BUG",
      },
    }
  }
>
  <div
    className="overview-quality-gate-condition-container display-flex-center"
  >
    <div
      className="overview-quality-gate-condition-value text-center spacer-right"
    >
      <Measure
        decimals={2}
        metricKey="new_reliability_rating"
        metricType="RATING"
        value="3"
      />
    </div>
    <div>
      <span
        className="overview-quality-gate-condition-metric little-spacer-right"
      >
        <IssueTypeIcon
          className="little-spacer-right"
          query="new_reliability_rating"
        />
        new_reliability_rating
      </span>
      <span
        className="little-spacer-top small text-muted"
      >
        quality_gates.operator.GT.rating
         
        A
      </span>
    </div>
  </div>
</Link>
`;

exports[`should render correclty 6`] = `
<Link
  className="overview-quality-gate-condition overview-quality-gate-condition-error"
  onlyActiveOnIndex={false}
  style={Object {}}
  to={
    Object {
      "pathname": "/project/issues",
      "query": Object {
        "id": "abcd-key",
        "resolved": "false",
        "severities": "BLOCKER,CRITICAL,MAJOR,MINOR",
        "sinceLeakPeriod": "true",
        "types": "VULNERABILITY",
      },
    }
  }
>
  <div
    className="overview-quality-gate-condition-container display-flex-center"
  >
    <div
      className="overview-quality-gate-condition-value text-center spacer-right"
    >
      <Measure
        decimals={2}
        metricKey="new_security_rating"
        metricType="RATING"
        value="3"
      />
    </div>
    <div>
      <span
        className="overview-quality-gate-condition-metric little-spacer-right"
      >
        <IssueTypeIcon
          className="little-spacer-right"
          query="new_security_rating"
        />
        new_security_rating
      </span>
      <span
        className="little-spacer-top small text-muted"
      >
        quality_gates.operator.GT.rating
         
        A
      </span>
    </div>
  </div>
</Link>
`;

exports[`should render correclty 7`] = `
<Link
  className="overview-quality-gate-condition overview-quality-gate-condition-error"
  onlyActiveOnIndex={false}
  style={Object {}}
  to={
    Object {
      "pathname": "/project/issues",
      "query": Object {
        "id": "abcd-key",
        "resolved": "false",
        "sinceLeakPeriod": "true",
        "types": "CODE_SMELL",
      },
    }
  }
>
  <div
    className="overview-quality-gate-condition-container display-flex-center"
  >
    <div
      className="overview-quality-gate-condition-value text-center spacer-right"
    >
      <Measure
        decimals={2}
        metricKey="new_maintainability_rating"
        metricType="RATING"
        value="3"
      />
    </div>
    <div>
      <span
        className="overview-quality-gate-condition-metric little-spacer-right"
      >
        <IssueTypeIcon
          className="little-spacer-right"
          query="new_maintainability_rating"
        />
        new_maintainability_rating
      </span>
      <span
        className="little-spacer-top small text-muted"
      >
        quality_gates.operator.GT.rating
         
        A
      </span>
    </div>
  </div>
</Link>
`;

exports[`should render correclty 8`] = `
<Link
  className="overview-quality-gate-condition overview-quality-gate-condition-error"
  onlyActiveOnIndex={false}
  style={Object {}}
  to={
    Object {
      "pathname": "/security_hotspots",
      "query": Object {
        "assignedToMe": undefined,
        "branch": undefined,
        "file": undefined,
        "hotspots": undefined,
        "id": "abcd-key",
        "pullRequest": undefined,
        "sinceLeakPeriod": undefined,
      },
    }
  }
>
  <div
    className="overview-quality-gate-condition-container display-flex-center"
  >
    <div
      className="overview-quality-gate-condition-value text-center spacer-right"
    >
      <Measure
        decimals={2}
        metricKey="security_hotspots_reviewed"
        metricType="RATING"
        value="3"
      />
    </div>
    <div>
      <span
        className="overview-quality-gate-condition-metric little-spacer-right"
      >
        <IssueTypeIcon
          className="little-spacer-right"
          query="security_hotspots_reviewed"
        />
        security_hotspots_reviewed
      </span>
      <span
        className="little-spacer-top small text-muted"
      >
        quality_gates.operator.GT.rating
         
        A
      </span>
    </div>
  </div>
</Link>
`;

exports[`should render correclty 9`] = `
<Link
  className="overview-quality-gate-condition overview-quality-gate-condition-error"
  onlyActiveOnIndex={false}
  style={Object {}}
  to={
    Object {
      "pathname": "/security_hotspots",
      "query": Object {
        "assignedToMe": undefined,
        "branch": undefined,
        "file": undefined,
        "hotspots": undefined,
        "id": "abcd-key",
        "pullRequest": undefined,
        "sinceLeakPeriod": "true",
      },
    }
  }
>
  <div
    className="overview-quality-gate-condition-container display-flex-center"
  >
    <div
      className="overview-quality-gate-condition-value text-center spacer-right"
    >
      <Measure
        decimals={2}
        metricKey="new_security_hotspots_reviewed"
        metricType="RATING"
        value="3"
      />
    </div>
    <div>
      <span
        className="overview-quality-gate-condition-metric little-spacer-right"
      >
        <IssueTypeIcon
          className="little-spacer-right"
          query="new_security_hotspots_reviewed"
        />
        new_security_hotspots_reviewed
      </span>
      <span
        className="little-spacer-top small text-muted"
      >
        quality_gates.operator.GT.rating
         
        A
      </span>
    </div>
  </div>
</Link>
`;

exports[`should work with branch 1`] = `
<Link
  className="overview-quality-gate-condition overview-quality-gate-condition-error"
  onlyActiveOnIndex={false}
  style={Object {}}
  to={
    Object {
      "pathname": "/project/issues",
      "query": Object {
        "branch": "branch-6.7",
        "id": "abcd-key",
        "resolved": "false",
        "sinceLeakPeriod": "true",
        "types": "CODE_SMELL",
      },
    }
  }
>
  <div
    className="overview-quality-gate-condition-container display-flex-center"
  >
    <div
      className="overview-quality-gate-condition-value text-center spacer-right"
    >
      <Measure
        decimals={2}
        metricKey="new_maintainability_rating"
        metricType="RATING"
        value="3"
      />
    </div>
    <div>
      <span
        className="overview-quality-gate-condition-metric little-spacer-right"
      >
        <IssueTypeIcon
          className="little-spacer-right"
          query="new_maintainability_rating"
        />
        new_maintainability_rating
      </span>
      <span
        className="little-spacer-top small text-muted"
      >
        quality_gates.operator.GT.rating
         
        A
      </span>
    </div>
  </div>
</Link>
`;
